modules/gapi/samples/pipeline_modeling_tool/pipeline.hpp

```diff
@@ -1,31 +1,41 @@
 #ifndef OPENCV_GAPI_PIPELINE_MODELING_TOOL_PIPELINE_HPP
 #define OPENCV_GAPI_PIPELINE_MODELING_TOOL_PIPELINE_HPP
 
+#include <iomanip>
+
 struct PerfReport {
-    std::string               name;
-    double  avg_latency       = 0.0;
-    double  throughput        = 0.0;
-    int64_t first_run_latency = 0;
-    int64_t elapsed           = 0;
-    int64_t compilation_time  = 0;
+    std::string name;
+    double  avg_latency        = 0.0;
+    int64_t min_latency        = 0;
+    int64_t max_latency        = 0;
+    int64_t first_latency      = 0;
+    double  throughput         = 0.0;
+    int64_t elapsed            = 0;
+    int64_t warmup_time        = 0;
+    int64_t num_late_frames    = 0;
     std::vector<int64_t> latencies;
 
     std::string toStr(bool expanded = false) const;
 };
 
 std::string PerfReport::toStr(bool expand) const {
     std::stringstream ss;
-    ss << name << ": Compilation time: " << compilation_time << " ms; "
-       << "Average latency: " << avg_latency << " ms; Throughput: "
-       << throughput << " FPS; First latency: "
-       << first_run_latency << " ms";
-
+    ss << name << ": \n"
+       << std::string(2, ' ') << "Warm up time: " << warmup_time << " ms\n"
+       << std::string(2, ' ') << "Execution time: " << elapsed << " ms\n"
+       << std::string(2, ' ') << "Frames:       " << num_late_frames << "/" << latencies.size()
+                              << " (late/all)\n"
+       << std::string(2, ' ') << "Latency:\n"
+       << std::string(4, ' ') << "first: " << first_latency << " ms\n"
+       << std::string(4, ' ') << "min:   " << min_latency   << " ms\n"
+       << std::string(4, ' ') << "max:   " << max_latency   << " ms\n"
+       << std::string(4, ' ') << "avg:   " << std::fixed << std::setprecision(3)
+                                           << avg_latency << " ms\n"
+       << std::string(2, ' ') << "Throughput: " << std::fixed << std::setprecision(3)
```

Comment:
Also..
Why couldn't you just put those extra spaces into the string literals? You have them anyway, and that way padding would have been seen in the code, too.